### PR TITLE
deploy: steps, petsc, petsc-py, and omega-h

### PR DESCRIPTION
**Pull request by tristan0x**
* remove dead config related to trilinos
* remove useless steps spec +petsc since it is now the default
* omega-h: remove module suffix "-gmsh"
* petsc:
	- turn off variant `hypre` which is enabled by default
	- provide 2 modules, with and without "complex" variant
* steps: provide modules versions
	- tag 4.0.0
	- virtual version 5.0.0a that point to a recent revision of master

### CODE REVIEW

File: var/spack/repos/builtin/packages/petsc/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/petsc/package.py:1]: Do we want to copy this recipe into `repo-patches` like for steps?

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -268,24 +268,17 @@ def check_fortran_compiler(self):
...
-    depends_on("hypre@develop+mpi~internal-superlu~int64", when="@main+hypre+mpi~complex~int64")
+    depends_on("hypre@:2.13~internal-superlu", when="@:3.8+hypre")
+    depends_on("hypre@2.14:2.18.2~internal-superlu", when="@3.9:3.13+hypre")
+    depends_on("hypre@2.14:2.22.0~internal-superlu", when="@3.14:3.15+hypre")
+    depends_on("hypre@2.14:~internal-superlu", when="@3.16:+hypre")
+    depends_on("hypre@develop~internal-superlu", when="@main+hypre")
+    depends_on("hypre+complex", when="+hypre+complex")
+    depends_on("hypre~complex", when="+hypre~complex")
+    depends_on("hypre+int64", when="+hypre+int64")
+    depends_on("hypre~int64", when="-hypre~int64")
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/petsc/package.py:279]:
```suggestion
    depends_on("hypre~int64", when="+hypre~int64")
```
? that `-` does not seem to make sense

File: bluebrain/deployment/environments/applications_hpc.yaml [hunk 2]
```diff
@@ -57,6 +52,7 @@ spack:
     - spatial-index
     - spykfunc
     - steps
```
Comment (matz-e) [inline bluebrain/deployment/environments/applications_hpc.yaml:55]: This steps now does not use `petsc`?

File: var/spack/repos/builtin/packages/petsc/package.py
Comment (tristan0x) [reply var/spack/repos/builtin/packages/petsc/package.py:1]: I don't know where we want to put this change in this repo. The only thing I know is that this change in PETSc recipe should be sent upstream eventually.
Let me know where you prefer to have this change (here, or copy in repo-patches)

File: bluebrain/deployment/environments/applications_hpc.yaml (hunk 2 above)
Comment (tristan0x) [reply bluebrain/deployment/environments/applications_hpc.yaml:55]: `+petsc` variant is the default in `steps` recipe. So both modules use PETSc. Maybe you prefer something more explicit like this?

```
- steps ^petsc~complex
- steps@5 ^petsc+complex
```

Comment (matz-e) [reply bluebrain/deployment/environments/applications_hpc.yaml:55]: Ah, ok. So then the package preference for `+petsc` was totally superfluous. Thanks for clarifying! Good as is, then.

File: var/spack/repos/builtin/packages/petsc/package.py
Comment (matz-e) [reply var/spack/repos/builtin/packages/petsc/package.py:1]: If these changes go into `builtin/petsc`, then filing a PR upstream for the same changes would be good.

**Comment by tristan0x** [comment]
Ready to merge this time @matz-e

**Review by matz-e — APPROVED** [review]

**Pull request merged**